src/hooks/useApplicationsInfo.ts

```diff
@@ -0,0 +1,35 @@
+import * as React from 'react';
+import { k8sGetResource } from '../dynamic-plugin-sdk';
+import { ApplicationModel } from '../models';
+import { useActiveNamespace } from './useActiveNamespace';
+
+type ApplicationInfo = {
+  loaded: boolean;
+  appExists: boolean;
+};
+
+export const useApplicationsInfo = (): ApplicationInfo => {
+  const namespace = useActiveNamespace();
+  const [applicationInfo, setApplicationInfo] = React.useState<ApplicationInfo>({
+    loaded: false,
+    appExists: false,
+  });
+
+  React.useEffect(() => {
+    let mounted = true;
+    const fetchApplication = async () => {
+      const appData = await k8sGetResource({ model: ApplicationModel, ns: namespace });
```

Comment:
Should only fetch application once namespace is loaded. Without namespace the first call fails because user cannot fetch applications at cluster scope.